en/flight_controller/crazyflie2.md

```diff
@@ -161,10 +163,24 @@ Then, you need to stick the battery on top of the SD card deck using a double si
 | IMU_DTERM_CUTOFF  | 70                          |
 | SYS_FMU_TASK      | Enabled                     |
 
-\* Since we haven't been able to fly it with optical flow, we don't use optical flow data yet.
+## Altitude Control 
 
+Crazyflie is able to fly in Altitude mode if you use a [Z-ranger deck](https://store.bitcraze.io/collections/decks/products/z-
+ranger-deck), connected to the board on I2C interface. The driver for this range finder is called "vl53lxx.cpp" which publishes 
+range readings on a uORB topic, named "distance_sensor". The maximum height (above ground) the range finder can sense lies in the 
+range of 0.5 to 1 m, depending on the reflective property of the ground surface. This means you cannot hold altitudes above this 
```

Comment:
Doesn't Craziflie have a barometer? That is usually used for these modes.

Reply:
Ah, I see it does, but you can't rely on it. Will the mode switch anyway, or is it disabled?

Reply:
Currently we are starting the baro driver anyway, therefore we can still switch to altitude but the performance is not good. We could consider of not starting the driver at all.

Reply:
@DanielePettenuzzo since the Baro is not reliable, I think we shouldn't start it at all.

@hamishwillee there's a different way of switching to Baro and this issue is still in discussion.

In short to switch to Baro, when range finder readings saturate, we:
* Set EKF2_HGT_MODE to Baro
* Enable EKF2_RNG_AID
* Set EKF2_RNG_A_HMAX  to 0.5 
* Comment out https://github.com/PX4/ecl/blob/master/EKF/control.cpp#L1054-L1072 to allow switching to Baro without the need of the flow deck.

Reply:
Users don't receive any warning. The crazyflie's altitude starts increasing and decreasing at mid-throttle above max height.